### PR TITLE
[Core Feature 3][Messaging]

### PR DESCRIPTION
**Pull request by andrea-io**
### Description
This messaging pull request contains the following **features**:
1. The ability for the user to select a user from the Detailed Profile VC (after filtering for users) and **start a message conversation** with them.
2. The ability for the user to select a user from the Detailed Profile VC (after filtering for users) and open a previously started **message conversation history**. 

> From the Conversation VC, a user can type their message into a text view and click a button to send the message to the other user. This message and any other message history appears on the user's view with the text sent and the username of the user who sent it.

3. The ability for the user to navigate to the Message Inbox VC (from the main Feed VC) and view **current conversations** in their inbox.
### Refactoring
I refactored my previous code to adjust the following:
- Removal of NSLog statements (Exception is for error blocks)
- Removal of commented out code
- Placed UserInfo-related functions into its own object class and removed them from the Conversation object class
- Removal of magic numbers and replaced with constants
### Demo

https://user-images.githubusercontent.com/66450436/182309671-e2e0ceef-75c2-49d5-8cf6-a1bdd94607f7.mov

https://user-images.githubusercontent.com/66450436/182309682-a47e13a0-a7be-4de3-8748-5cd98e021d48.mov

https://user-images.githubusercontent.com/66450436/182309686-ee066577-02b7-4bf7-9cf6-c04c414806bc.mov

### CODE REVIEW

File: ShareTheTable/View Controllers/ConversationViewController.m [hunk 1]
```diff
@@ -18,40 +20,52 @@ @implementation ConversationViewController
...
 
 - (void)onTimer {
     [self refreshConversationData];
 }
 
-- (void)refreshConversationData {
-    Conversation* convo = [[Conversation alloc] init];
-    NSLog(@"Other user: %@", self.otherUser);
+- (void)initializeUserData {
+    // Repeated code below, will create a function to fetch UserInfo
```
Comment (Ernest1418) [inline ShareTheTable/View Controllers/ConversationViewController.m:38]: Great job calling out opportunities for refactoring code

File: ShareTheTable/View Controllers/Profile Making/Question/QuestionViewController.m [hunk 2]
```diff
@@ -18,14 +18,16 @@ @interface QuestionViewController ()
...
+NSInteger const DIET_PREF_TAG = 1;
+NSInteger const FOOD_PREF_TAG = 2;
+
 - (IBAction)didTapAButton:(id)sender {
     UIButton *button = (UIButton *)sender;
     // Refresh button selected
     [self refreshButtons:button];
-    NSLog(@"Button pressed: %@", [sender currentTitle]);
     
     // Tag = 1, button pressed was a dietary preference
```
Comment (Ernest1418) [inline ShareTheTable/View Controllers/Profile Making/Question/QuestionViewController.m:29]: Nit: This comment is unnecessary now

File: ShareTheTable/View Controllers/Profile Making/Question/QuestionViewController.m [hunk 3]
```diff
@@ -18,14 +18,16 @@ @interface QuestionViewController ()
...
-    if([button tag] == 1) {
+    if([button tag] == DIET_PREF_TAG) {
         PFObject* dietPreference = [[PFObject alloc] initWithClassName:@"Preference"];
         // Check if the button was selected or deselected
```
Comment (Ernest1418) [inline ShareTheTable/View Controllers/Profile Making/Question/QuestionViewController.m:32]: This comment is unnecessary as the code shows the same thing

File: ShareTheTable/View Controllers/Profile Making/Question/QuestionViewController.m [hunk 4]
```diff
@@ -68,7 +70,7 @@ - (IBAction)didTapAButton:(id)sender {
             }];
         }
     // Tag = 2, button pressed was a food preference
-    } else if([button tag] == 2) {
+    } else if([button tag] == FOOD_PREF_TAG) {
         PFObject* foodPreference = [[PFObject alloc] initWithClassName:@"Preference"];
```
Comment (Ernest1418) [inline ShareTheTable/View Controllers/Profile Making/Question/QuestionViewController.m:74]: Unnecessary comment

**Review by Ernest1418 — COMMENTED** [review]
Great job incorporating feedback from previous pull requests.

File: ShareTheTable/Models/Conversation.h [hunk 5]
```diff
@@ -6,19 +6,18 @@
...
-#import "UserInfo.h"
+@class UserInfo;
+
 NS_ASSUME_NONNULL_BEGIN
 
 @interface Conversation : PFObject<PFSubclassing>
 
-@property (nonatomic, strong) PFObject* userOnePoint;
-@property (nonatomic, strong) PFObject* userTwoPoint;
+@property (nonatomic, strong) PFObject* userInfoOnePointer;
```
Comment (anishakollareddy) [inline ShareTheTable/Models/Conversation.h:15]: nit: better variable names, should this be userOneInfo/userTwoInfo?

**Review by anishakollareddy — COMMENTED** [review]
Great job with addressing past comments, the refactor looks good!

**Pull request merged**